woocommerce/wc-admin
# Update loopkup tables when an order is deleted

Fixes #1307

The lookup tables are not affected when an order is deleted causing incorrect stats. Each data store should contain information about how to remove data when deletion occurs.

### Detailed test instructions:

1. Create an order with products, taxes and coupons.
2. You can check that data is created in `wc_order_stats`, `wc_order_coupon_lookup`, `wc_order_product_lookup`, and `wc_order_tax_lookup`.
3. Delete your order created in step 1.
4. Check your database, note that all order data is only deleted from `wc_order_stats`, but still in the other tables.
5. Apply this patch and repeat steps 1, 2 and 3.
6. Check that all order data is cleaned from all lookup tables.

Note: There's some code duplication to delete order, still I think that is best to control it in each data store, instead of handing all inside `WC_Admin_Reports_Orders_Stats_Data_Store::delete_order`.
